x/gov/types/v1/params.go

```diff
@@ -419,5 +423,16 @@ func (p Params) ValidateBasic() error {
 		return fmt.Errorf("minimum initial deposit decrease ratio too large: %s", minInitialDepositDecreaseRatio)
 	}
 
+	burnDepositNoThreshold, err := sdk.NewDecFromStr(p.BurnDepositNoThreshold)
```

Comment:
We need to add a check for threshold, law and amendment threshold

so that the `burnDepositNoThreshold > 1 - threshold` (or the burn would happen with less than no votes to make a proposal fail, which makes no sense since the proposal would mean is passing)

the same check goes for law and amendment thresholds

Reply:
If the proposal is passing, the `burnDepositNoThreshold` is not checked, so the burn would not happen anyway. But I will still do this `burnDepositNoThreshold > 1 - threshold` check so the condition is not tied to the order of code execution.

Reply:
done here 3788dbb7